### ISSUE TITLE
Collecting an AxisArray produces an Array

### ISSUE THREAD

**Issue by grahamas** [Tokazama/AxisIndices.jl#41]
It's not obvious what this *should* do, but I was surprised. My my instinct is that this should collect the parent, which would also functionally resolve #34.
```
julia> using AxisIndices; A = AxisArray(rand(6, 5), 0.5:0.1:1.0, 1:5.);

julia> collect(A)
6×5 Array{Float64,2}:
 0.0127804  0.44732    0.269172  0.177361  0.996044
 0.0278617  0.477912   0.526784  0.483331  0.498996
 0.173871   0.847214   0.569129  0.157995  0.957401
 0.0824858  0.196337   0.121123  0.903437  0.636979
 0.139283   0.0874877  0.225654  0.755756  0.677682
 0.684797   0.507165   0.408826  0.123632  0.168082

julia> which(collect, (AbstractAxisArray,))
collect(A::AbstractArray) in Base at array.jl:626
```

**Comment by Tokazama** [comment]
The doc string for `collect` says "Return an `Array` of all items in a collection or iterator.", so it does specifically it returns an `Array` type. However, I'm not against changing this to call `collect` on the parent data and then rewrap the axes. It would just need to be documented because it does deviate from original doc string.

**Comment by Tokazama** [comment]
Resolved with fee9392
